node/pkg/processor/observation.go

```diff
@@ -306,8 +306,13 @@ func (p *Processor) handleInboundSignedVAAWithQuorum(m *gossipv1.SignedVAAWithQu
 		return
 	}
 
+	addresses := p.gs.Keys
```

Comment:
I don't think it is too odd. 
the verifier should know which keys to verify against, and if it vaaV1 - then it'll take the keys of the guardians, otherwise it'll take the key of the TSS.